lib/dsprot/include/asm_macro.inc

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include "encoding_constants.h"
+
+	.macro arm_func_start name
```

Comment:
pretty sure we already have these macros somewhere
probably a good idea to use them instead of redefining them

Reply:
tbh I dislike this
I mean... I guess I get why but if other libs do this too then this is something that should be addressed for all libs and not in this specific PR